django/db/backends/mysql/operations.py

```diff
@@ -199,11 +199,12 @@ def bulk_insert_sql(self, fields, placeholder_rows):
         return "VALUES " + values_sql
 
     def combine_expression(self, connector, sub_expressions):
-        """
-        MySQL requires special cases for ^ operators in query expressions
-        """
         if connector == '^':
             return 'POW(%s)' % ','.join(sub_expressions)
+        # MySQL's binary operators return an unsigned 64-bit integer.
+        # We convert result back to signed integer.
+        elif connector in ['&', '|']:
```

Comment:
It's quite interesting how Python handles the two cases under the hood. Since tuples are immutable, the whole object can be cached as a 'const' variable for the function, but as lists aren't, it must reconstruct the object (from const 1 and 2) on each invocation. You can see this using the `dis` module and looking at the bytecode:

```
In [1]: import dis

In [2]: def foo():
   ...:     [1, 2]
   ...:

In [3]: def bar():
   ...:     (1, 2)
   ...:

In [4]: dis.dis(foo)
  2           0 LOAD_CONST               1 (1)
              2 LOAD_CONST               2 (2)
              4 BUILD_LIST               2
              6 POP_TOP
              8 LOAD_CONST               0 (None)
             10 RETURN_VALUE

In [5]: dis.dis(bar)
  2           0 LOAD_CONST               3 ((1, 2))
              2 POP_TOP
              4 LOAD_CONST               0 (None)
              6 RETURN_VALUE
```

Anyway, it's mostly a micro optimization, but if it's always faster, it can't hurt!